terraform/P3/5.0-build-frontend.tf

```diff
@@ -5,15 +5,38 @@ resource "local_file" "api_config" {
   EOT
 }
 
+resource "null_resource" "install_nodejs" {
+  depends_on = [local_file.api_config]
+
+  provisioner "local-exec" {
+    command = <<-EOT
+      # Check if Node.js is already installed
+      if (-not (Test-Path "$env:ProgramFiles\nodejs\node.exe")) {
+        # Download and install Node.js
+        winget install --uninstall-previous OpenJS.NodeJS
+      } else {
+          Write-Host "Node.js is already installed."
+      }
+    EOT
+
+    interpreter = ["PowerShell", "-Command"]
+  }
+}
+
 resource "null_resource" "build_frontend" {
+  depends_on = [null_resource.install_nodejs]
   triggers = {
     build_trigger = "../../frontend/.env.production"
   }
 
   provisioner "local-exec" {
-    command = "npm install vite@latest && npm install npm install && npm run build"
+    command = <<-EOT
+      if (-not $env:Path.Contains('C:\\Program Files\\nodejs')) {
+        $env:Path += ';C:\\Program Files\\nodejs'
+      }
+      & 'C:\\Program Files\\nodejs\\npm.cmd' install ; & 'C:\\Program Files\\nodejs\\npm.cmd' install vite@latest ; & 'C:\\Program Files\\nodejs\\npm.cmd' run build
+    EOT
```

Comment:
Y'a juste un monde où si son DD c'est pas sur C:\\ la commande foire
Mais je pense que ce monde est inéxistant